python_modules/dagster/dagster/core/execution/plan/outputs.py

```diff
@@ -22,6 +23,7 @@ class StepOutputProperties(
             ("is_dynamic", bool),
             ("is_asset", bool),
             ("should_materialize", bool),
+            ("asset_key", Optional[AssetKey]),
```

Comment:
its a pain to backcompat but should asset_key subsume / deprecate is_asset? Its information duplication right or am i missing something?

Reply:
In most cases the information is the same, but for dynamic asset keys `is_asset` will return `True`, but `asset_key` will only return a value for hardcoded asset keys. We plan on deprecating dynamic asset keys by 0.15.0

Reply:
I think that when we fully deprecate dynamic asset keys we can deprecate `is_asset` and just use `asset_key`.